outlook/src/taskpane/components/Log/Logger.tsx

```diff
@@ -27,21 +34,141 @@ class Logger extends React.Component<LoggerProps, LoggerState> {
+                    resolve({
+                        name: attachment.name, inline: true,
+                        oversize: true, index
+                    });
+                } else {
+                    resolve({
+                        name: attachment.name, inline: false,
+                        oversize: true, index
+                    });
+                }
+            }
+            Office.context.mailbox.item.getAttachmentContentAsync(attachment.id, asyncResult => {
+                if (attachment.isInline && attachment.contentType.indexOf("image") >= 0) {
+                    resolve({
+                        name: attachment.name,
+                        content: asyncResult.value.content,
+                        inline: true,
+                        index: index,
+                    });
+                } else {
+                    resolve({
+                        name: attachment.name,
+                        content: asyncResult.value.content,
+                        inline: false,
+                        index: index,
+                    });
+                }
+            });
+        });
+    };
+
+    private logRequest = async (event): Promise<any> => {
         event.stopPropagation();
 
         this.setState({logged: 1});
-        Office.context.mailbox.item.body.getAsync(Office.CoercionType.Html, (result) => {
+        Office.context.mailbox.item.body.getAsync(Office.CoercionType.Html, async (result) => {
             const msgHeader = '<div>From: ' + Office.context.mailbox.item.sender.emailAddress + '</div><br/>';
-            const msgFooter = '<br/><div class="text-muted font-italic">Logged from <a href="https://www.odoo.com/documentation/user/crm/optimize/mail_client_extension.html" target="_blank">Outlook Inbox</a></div>';
+            const msgFooter = '<br/><div class="text-muted font-italic">Logged from <a href="https://www.odoo.com/documentation/14.0/applications/sales/crm/optimize/outlook_extension.html" target="_blank">Outlook Inbox</a></div>';
             const body = result.value.split('<div id="x_appendonsend"></div>')[0]; // Remove the history and only log the most recent message.
             const message = msgHeader + body + msgFooter;
 
             const requestJson = {
                 res_id: this.props.resId,
                 model: this.props.model,
-                message: message
+                message: message,
+                attachments: [],
+            }
+
+            const doc = new DOMParser().parseFromString(requestJson.message, "text/html");
+
+            const officeAttachmentDetails = Office.context.mailbox.item.attachments;
+
+            let size = 0;
+
+            //check if attachment size is bigger then the threshold
+            officeAttachmentDetails.forEach(officeAttachment => {
+                size += officeAttachment.size
+            });
+
+            const promises: any[] = [];
+
+            if (size > totalSizeThreshold * 1024 * 1024) {
```

Comment:
Mmmh now that I'm seeing it being used I think the constant should already contain this computation.

So 
```js
// total attachments size threshold in bytes
const SIZE_THRESHOLD_TOTAL = 40 * 1024 * 1024;
```

Makes it a bit easier to use.

Reply:
The issue here is that we still need a value in MB to show in the warning message, for this reason, I am proposing to have two constants one for MB and one for Bytes, like this:

```
//total attachments size threshold
const SIZE_THRESHOLD_TOTAL_MB = 40;
const SIZE_THRESHOLD_TOTAL_BYTES = SIZE_THRESHOLD_TOTAL_MB * 1024 * 1024;

//single attachment size threshold
const SIZE_THRESHOLD_SINGLE_ELEMENT_MB = 10;
const SIZE_THRESHOLD_SINGLE_ELEMENT_BYTES = SIZE_THRESHOLD_SINGLE_ELEMENT_MB * 1024 * 1024;
```

Can you tell me what do you think about it?

Reply:
Hmm yes I see.
Then only keep the constant in MB, having four constants seems a bit much.

Sorry for the back and forth.

Reply:
Okay, no problem, going back to the previous implementation.